src/library/scanner/libraryscanner.cpp

```diff
@@ -422,27 +428,63 @@ void LibraryScanner::slotFinishUnhashedScan() {
         kLogger.debug() << "Scan cancelled";
     }
 
-    // TODO(XXX) doesn't take into account verifyRemainingTracks.
-    qDebug("Scan took: %s. "
-           "%d unchanged directories. "
-           "%d changed/added directories. "
-           "%d tracks verified from changed/added directories. "
-           "%d new tracks.",
-            m_scannerGlobal->timerElapsed().formatNanosWithUnit().toLocal8Bit().constData(),
-            static_cast<int>(m_scannerGlobal->verifiedDirectories().size()),
-            m_scannerGlobal->numScannedDirectories(),
-            static_cast<int>(m_scannerGlobal->verifiedTracks().size()),
-            static_cast<int>(m_scannerGlobal->addedTracks().size()));
+    const QString durationString = m_scannerGlobal->timerElapsed().formatMillisWithUnit();
+    const int numVerifiedDirs = static_cast<int>(m_scannerGlobal->verifiedDirectories().size());
+    const int numScannedDirs = m_scannerGlobal->numScannedDirectories();
+    const int numVerifiedTracks = static_cast<int>(m_scannerGlobal->verifiedTracks().size());
+    const int numNewTracks = m_scannerGlobal->addedTracks().size() - m_numRelocatedTracks;
+
+    const QSet<QString> existingTracks = m_trackDao.getAllExistingTrackLocations();
+    int numRediscoveredTracks = 0;
+    for (const QString& loc : std::as_const(m_previouslyMissingTracks)) {
+        if (existingTracks.contains(loc)) {
+            numRediscoveredTracks++;
+        }
+    }
+    const auto missingTracks = m_trackDao.getAllMissingTrackLocations();
```

Comment:
I have no idea how this performs and whether optimization is beneficial:
An idea for optimizion would be to move the compare loop into getAllMissingTrackLocations and get around the temporary QSet(). This will require a lot more compares though.